### PR TITLE
Hbba battery

### PR DESCRIPTION
**Pull request by Nick-GB**

### CODE REVIEW

**Comment by Nick-GB** [comment]
@AlexCampanozzi

File: homodeus_behaviors/scripts/bhvr_docking.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        self.undock_client = actionlib.SimpleActionClient("undocker_server", UndockAction)
+        self.dock_client.wait_for_server()
+        self.dock_client.send_goal(goal)
+
+        
+    def is_docked_cb(self, is_docked):
+        if (is_docked.data):
+            self.dock_checker_sub.unregister()
+            rospy.loginfo("the robot is docked!")
+            quit()
```
Comment (JMrobotUdes) [inline homodeus_behaviors/scripts/bhvr_docking.py:38]: We do not wan to quit the node if the robot's docked!?

File: homodeus_behaviors/scripts/bhvr_docking.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+from dock_charge_sm_msgs.msg import GoAndDockAction, GoAndDockGoal
+from laser_servoing_msgs.msg import UndockAction, UndockGoal
+
+from std_msgs.msg import Bool
+from base_navigation.scripts.navigator import Navigator
+
+class Docking():
+    def __init__ ( self ) :
+        self.dock_checker_sub = rospy.Subscriber("/power/is_docked", Bool, self.is_docked_cb)
+        self.is_docked = False
```
Comment (JMrobotUdes) [inline homodeus_behaviors/scripts/bhvr_docking.py:15]: it's never used

File: homodeus_behaviors/scripts/bhvr_docking.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+
+if __name__ == '__main__':
+    try:
+        rospy.init_node('bhvr_docking')
+        
+        mode = "dock"
+        docking = Docking()
+
+        if mode == "dock":
```
Comment (JMrobotUdes) [inline homodeus_behaviors/scripts/bhvr_docking.py:49]: Is that only a test for docking, there is no logic at all in this node and it is node reusable

File: homodeus_hbba_cfg/hbba_cfg/bhvr_docking.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    custom_bringup: "pubNavGoal('/hbba_nav_goal', params.frame_id, params.x, params.y, params.t);"
+    
+- strat:
+    name: TIAGoToLandmark
+    class: GoToLandmark
+    utility: 8
+    costs: {move_stack: 1}
+    modules:
+        - tiago_nav
+    custom_bringup: "pubString('/hbba_landmark_nav_goal', params.name);"
```
Comment (JMrobotUdes) [inline homodeus_hbba_cfg/hbba_cfg/bhvr_docking.yaml:31]: the strategies has no link with docking behavior..... nothing is good in this review

File: homodeus_perceptions/scripts/tester_battery_level.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        rospy.init_node(common.get_file_name(__file__))
+        node = Tester_battery_Level()
+        node.fake_batt_lev()
+
+        rospy.on_shutdown(node.node_shutdown)
+        
+        rospy.spin()
+
+    except Exception:
+        common.logerr(__file__,traceback.format_exc())
```
Comment (JMrobotUdes) [inline homodeus_perceptions/scripts/tester_battery_level.py:44]: This isn't useful at all..

File: homodeus_perceptions/scripts/proc_battery_level.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        #self.input_perc = rospy.Subscriber("/power_status/charge", Float32, self.perc_to_level)
+        self.input_perc = rospy.Subscriber("fake_battery_percent", Float32, self.perc_to_level) #use for test,put power_status/charge in comment
+        self.level = "high"
+
+    def transform(self):
+        """
+        This method transform the input which is a percentage into a battery level which is a string
+        """
+        while not rospy.is_shutdown():
+            rate = rospy.Rate(10)
```
Comment (JMrobotUdes) [inline homodeus_perceptions/scripts/proc_battery_level.py:29]: no need of rospy.Rate

**Review by JMrobotUdes — COMMENTED** [review]
I think the best solution is simply to restart this part of the project. It isn't useful and has many flaws. We'll restart from the master.

**Comment by Nick-GB** [comment]
Yes this was just to test the docking feature, but I agree that we can start from scratch (using the core ideas of this code) if we feel we ever need this behavior.

**Pull request closed without merging**